modules/custom/az_event/src/Plugin/views/filter/AZCalendarFilter.php

```diff
@@ -0,0 +1,172 @@
+      }
+
+      $this->view->element['#attached']['library'][] = 'az_event/az_calendar_filter';
+      $this->view->element['#attached']['drupalSettings']['azCalendarFilter'] = $filter_settings;
+      $this->view->element['#cache']['max-age'] = 0;
+      // Prepare a wrapper for the calendar JS to access.
+      $calendar_element = [
+        '#type' => 'container',
+        '#attached' => ['library' => ['az_event/az_calendar_filter']],
+        '#attributes' => [
+          'class' => [
+            'az-calendar-filter-wrapper',
+          ],
+          // Communicate to the HTML DOM what the unique id of our filter is.
+          'data-az-calendar-filter' => $this->options['expose']['identifier'],
+        ],
+      ];
+
+      // Add the calendar markup into the widget.
+      array_unshift($form['value'], $calendar_element);
+      $form['value']['#attributes']['class'][] = 'views-widget-az-calendar-filter';
+      $form['value']['#type'] = 'container';
+    }
+  }
+
+  /**
+   * Clones the view and returns calendar cell preview.
+   *
+   * @return array
+   *   The preview cells.
+   */
+  protected function calendarCells() {
+    $cells = [];
+    if (empty($this->view)) {
+      return $cells;
+    }
+    $view = Views::getView($this->view->id());
+    $view = Views::executableFactory()->get($this->view->storage);
+    $view->cellQuery = TRUE;
+
+    if (empty($view)) {
+      return $cells;
+    }
+    $view->setDisplay($this->view->current_display);
+
+    // Turn off the pager for the cell query.
+    $pager = $view->display_handler->getOption('pager');
+    $pager['type'] = 'none';
+    $view->display_handler->setOption('pager', $pager);
+
+    // Copy over exposed input.
+    $input = $this->view->getExposedInput();
+    $view->setExposedInput($input);
+
+    // Copy over arguments.
+    $args = $this->view->args;
+    $view->setArguments($args);
+
+    $view->execute();
+
+    foreach ($view->result as $index => $row) {
+      if (!empty($row->az_calendar_filter_start) && !empty($row->az_calendar_filter_end)) {
+        $cells[] = [
+          $row->az_calendar_filter_start,
+          $row->az_calendar_filter_end,
+        ];
+      }
+    }
+
+    $view->cellQuery = FALSE;
+    $view->destroy();
+    return $cells;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    $this->ensureMyTable();
+    $field = "$this->tableAlias.$this->realField" . '_value';
+
+    $info = $this->operators();
+    if (!empty($info[$this->operator]['method'])) {
+      $this->{$info[$this->operator]['method']}($field);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function operators() {
+    $operators = [
+      'overlaps' => [
+        'title' => $this->t('Overlaps'),
+        'method' => 'opOverlap',
+        'short' => $this->t('overlaps'),
+        'values' => 2,
+      ],
+    ];
+
+    return $operators;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
```

Comment:
Nitpick.  It'd probably be good to add a description here since it doesn't look like this method is provided by a parent class (AFAICT).